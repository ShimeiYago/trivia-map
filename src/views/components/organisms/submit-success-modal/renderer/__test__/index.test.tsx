import { shallow, ShallowWrapper } from 'enzyme';
import { Renderer, State, Props } from '..';

let shallowWrapper: ShallowWrapper<Props, State, Renderer>;

const props: Props = {
  articleFormSubmittingState: 'waiting',
  submitSuccessInfo: { postId: 1, title: 'title', description: 'description' },
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    shallowWrapper = shallow(<Renderer {...props} />);
  });

  it('basic', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('without submitSuccessInfo', () => {
    shallowWrapper.setProps({
      submitSuccessInfo: undefined,
    });
    expect(shallowWrapper).toMatchSnapshot();
  });
});

describe('componentDidUpdate', () => {
  beforeEach(() => {
    shallowWrapper = shallow(<Renderer {...props} />);
  });

  it('should change states to show post success message', () => {
    shallowWrapper.setProps({
      articleFormSubmittingState: 'success',
    });
    const instance = shallowWrapper.instance();

    instance.componentDidUpdate({
      ...props,
      articleFormSubmittingState: 'loading',
    });
    expect(instance.state.show).toBeTruthy();
  });
});

describe('handleCloseMessage', () => {
  it('should change show state', () => {
    shallowWrapper = shallow(<Renderer {...props} />);
    shallowWrapper.setState({ show: true });
    const instance = shallowWrapper.instance();

    instance['handleCloseMessage']();
    expect(instance.state.show).toBeFalsy();
  });
});
