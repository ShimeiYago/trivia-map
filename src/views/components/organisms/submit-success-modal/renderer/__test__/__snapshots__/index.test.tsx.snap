// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<BoxModal
  onClose={[Function]}
  open={false}
  showCloseButton={true}
>
  <ForwardRef(Stack)
    pb={3}
    px={2}
    spacing={2}
  >
    <ForwardRef(Typography)
      align="center"
      variant="h5"
    >
      投稿が完了しました！
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      align="center"
    >
      <Link
        to="/articles/1"
      >
        title
      </Link>
    </ForwardRef(Typography)>
    <ForwardRef(Box)>
      <ForwardRef(Typography)
        align="center"
        variant="h6"
      >
        SNSでシェアしませんか？
      </ForwardRef(Typography)>
      <ShareButtons
        description="description"
        title="title"
        url="http://localhost/articles/1"
      />
    </ForwardRef(Box)>
  </ForwardRef(Stack)>
</BoxModal>
`;

exports[`Shallow Snapshot Tests without submitSuccessInfo 1`] = `""`;
